consensus/snapshot/snapshot.c

```diff
@@ -76,9 +81,123 @@ static retcode_t iota_snapshot_initial_state(snapshot_t *const snapshot,
  * Public functions
  */
 
+retcode_t iota_snapshot_init_conf(char const *const snapshot_conf_file,
+                                  snapshot_conf_t *const conf, bool testnet) {
+  retcode_t ret = RC_OK;
+  FILE *file = NULL;
+  size_t len = 0;
+  char *content = NULL;
+  cJSON *json = NULL, *tmp = NULL, *timestamp = NULL, *signature = NULL,
+        *coordinator = NULL;
+
+  if ((file = fopen(snapshot_conf_file, "r")) == NULL) {
+    log_error(SNAPSHOT_LOGGER_ID, "Snapshot configuration file not found\n");
+    ret = RC_SNAPSHOT_FILE_NOT_FOUND;
+    goto done;
+  }
+
+  if (fseek(file, 0, SEEK_END) < 0 || (len = ftell(file)) < 0 ||
+      fseek(file, 0, SEEK_SET) < 0) {
+    log_error(SNAPSHOT_LOGGER_ID, "Invalid snapshot configuration file\n");
+    ret = RC_SNAPSHOT_INVALID_FILE;
+    goto done;
+  }
+
+  if ((content = (char *)malloc(len + 1)) == NULL) {
+    ret = RC_SNAPSHOT_OOM;
+    goto done;
+  }
+
+  if (fread(content, 1, len, file) < 0) {
+    log_error(SNAPSHOT_LOGGER_ID, "Invalid snapshot configuration file\n");
+    ret = RC_SNAPSHOT_INVALID_FILE;
+    goto done;
+  }
+
+  content[len] = '\0';
+
+  if ((json = cJSON_Parse(content)) == NULL) {
+    goto json_error;
+  }
+
+  if ((timestamp = cJSON_GetObjectItemCaseSensitive(json, "timestamp")) ==
+          NULL ||
+      (coordinator = cJSON_GetObjectItemCaseSensitive(json, "coordinator")) ==
+          NULL) {
+    goto json_error;
+  }
+
+  if (!cJSON_IsNumber(timestamp)) {
+    goto json_error;
+  }
+  conf->timestamp = timestamp->valueint;
+
+  tmp = cJSON_GetObjectItemCaseSensitive(coordinator, "pubkey");
+  if (tmp == NULL || !cJSON_IsString(tmp) || tmp->valuestring == NULL ||
+      strlen(tmp->valuestring) != HASH_LENGTH_TRYTE) {
+    goto json_error;
+  }
+  memcpy(conf->coordinator, tmp->valuestring, HASH_LENGTH_TRYTE);
+
+  tmp = cJSON_GetObjectItemCaseSensitive(coordinator, "lastMilestone");
+  if (tmp == NULL || !cJSON_IsNumber(tmp)) {
+    goto json_error;
+  }
+  conf->last_milestone = tmp->valueint;
+
+  if (!testnet) {
+    if ((signature = cJSON_GetObjectItemCaseSensitive(json, "signature")) ==
+        NULL) {
+      goto json_error;
+    }
+
+    tmp = cJSON_GetObjectItemCaseSensitive(signature, "index");
+    if (tmp == NULL || !cJSON_IsNumber(tmp)) {
+      goto json_error;
+    }
+    conf->signature_index = tmp->valueint;
+
+    tmp = cJSON_GetObjectItemCaseSensitive(signature, "depth");
+    if (tmp == NULL || !cJSON_IsNumber(tmp)) {
+      goto json_error;
+    }
+    conf->signature_depth = tmp->valueint;
+
+    tmp = cJSON_GetObjectItemCaseSensitive(signature, "pubkey");
+    if (tmp == NULL || !cJSON_IsString(tmp) || tmp->valuestring == NULL ||
+        strlen(tmp->valuestring) != HASH_LENGTH_TRYTE) {
+      goto json_error;
+    }
+    memcpy(conf->signature_pubkey, tmp->valuestring, HASH_LENGTH_TRYTE);
+  }
+
+  goto done;
+
+json_error : {
```

Comment:
any reason for the braces?

Reply:
Labels can only point to statements or blocks.
So you have to put braces or do something like `json_error : ;` if you want to declare variables after a label.